packages/twenty-front/src/modules/workflow/workflow-trigger/components/WorkflowEditTriggerCronForm.tsx

```diff
@@ -0,0 +1,291 @@
+import { WorkflowCronTrigger } from '@/workflow/types/Workflow';
+import { useIcons } from 'twenty-ui';
+import { useTheme } from '@emotion/react';
+import { Select } from '@/ui/input/components/Select';
+import { WorkflowStepHeader } from '@/workflow/workflow-steps/components/WorkflowStepHeader';
+import { WorkflowStepBody } from '@/workflow/workflow-steps/components/WorkflowStepBody';
+import { getTriggerIcon } from '@/workflow/workflow-trigger/utils/getTriggerIcon';
+import { isDefined } from 'twenty-shared';
+import { getTriggerDefaultLabel } from '@/workflow/workflow-trigger/utils/getTriggerLabel';
+import { CRON_TRIGGER_INTERVAL_OPTIONS } from '@/workflow/workflow-trigger/constants/CronTriggerIntervalOptions';
+import cron from 'cron-validate';
+import { useState } from 'react';
+import { FormTextFieldInput } from '@/object-record/record-field/form-types/components/FormTextFieldInput';
+import { FormNumberFieldInput } from '@/object-record/record-field/form-types/components/FormNumberFieldInput';
+import { isNumber } from '@sniptt/guards';
+import { getCronTriggerDefaultSettings } from '@/workflow/workflow-trigger/utils/getCronTriggerDefaultSettings';
+
+type WorkflowEditTriggerCronFormProps = {
+  trigger: WorkflowCronTrigger;
+  triggerOptions:
+    | {
+        readonly: true;
+        onTriggerUpdate?: undefined;
+      }
+    | {
+        readonly?: false;
+        onTriggerUpdate: (trigger: WorkflowCronTrigger) => void;
+      };
+};
+
+type FormErrorMessages = {
+  CUSTOM?: string | undefined;
+  HOURS_hour?: string | undefined;
+  HOURS_minute?: string | undefined;
+  MINUTES?: string | undefined;
+};
+
+export const WorkflowEditTriggerCronForm = ({
+  trigger,
+  triggerOptions,
+}: WorkflowEditTriggerCronFormProps) => {
+  const theme = useTheme();
+  const [errorMessages, setErrorMessages] = useState<FormErrorMessages>({});
+  const [errorMessagesVisible, setErrorMessagesVisible] = useState(false);
+
+  const { getIcon } = useIcons();
+
+  const headerIcon = getTriggerIcon({
+    type: 'CRON',
+  });
+
+  const defaultLabel =
+    getTriggerDefaultLabel({
+      type: 'CRON',
+    }) ?? '';
+
+  const headerTitle = isDefined(trigger.name) ? trigger.name : defaultLabel;
+
+  const headerType = 'Trigger';
+
+  const onBlur = () => {
+    setErrorMessagesVisible(true);
+  };
+
```

Comment:
There are several validation patterns. React Hook Form implements these (https://react-hook-form.com/docs/useform#mode):

![CleanShot 2025-02-07 at 17 13 03@2x](https://github.com/user-attachments/assets/69f35098-b691-4171-b5fb-eda58e95edb9)

You implemented the `onTouched` behavior here, which is absolutely correct. We'll have to confirm that this is the behavior we want globally. Let's discuss it later!